### ISSUE TITLE
Latest commit breaks reuse of kinesalite object

### ISSUE THREAD

**Issue by rclark** [mhart/kinesalite#10]
After [the lastest commit](https://github.com/mhart/kinesalite/commit/1909053a5408672316ab1c774f8629847ba3c2f6), the following little script fails with
```
/Users/r/tmp/node_modules/kinesalite/node_modules/level-sublevel/nut.js:196
        return wrapIterator((db.db || db).iterator(opts))
                                          ^
TypeError: Object LevelUP has no method 'iterator'
    at Object.iterator (/Users/r/tmp/node_modules/kinesalite/node_modules/level-sublevel/nut.js:196:43)
    at EventEmitter.emitter.readStream.emitter.createReadStream (/Users/r/tmp/node_modules/kinesalite/node_modules/level-sublevel/shell.js:137:18)
    at EventEmitter.emitter.valueStream.emitter.createValueStream (/Users/r/tmp/node_modules/kinesalite/node_modules/level-sublevel/shell.js:152:20)
    at Object.sumShards (/Users/r/tmp/node_modules/kinesalite/db/index.js:249:29)
    at /Users/r/tmp/node_modules/kinesalite/actions/createStream.js:20:10
    at /Users/r/tmp/node_modules/kinesalite/node_modules/level-sublevel/shell.js:101:15
    at /Users/r/tmp/node_modules/kinesalite/node_modules/level-sublevel/nut.js:120:19
    at dispatchError (/Users/r/tmp/node_modules/kinesalite/node_modules/levelup/lib/util.js:77:7)
    at maybeError (/Users/r/tmp/node_modules/kinesalite/node_modules/levelup/lib/levelup.js:169:5)
    at LevelUP.get (/Users/r/tmp/node_modules/kinesalite/node_modules/levelup/lib/levelup.js:202:7)
```

```js
var kinesalite = require('kinesalite')({ createStreamMs: 0, deleteStreamMs: 0, ssl: false });
var AWS = require('aws-sdk');
var client = new AWS.Kinesis({
    region: 'fake',
    accessKeyId: 'fake',
    secretAccessKey: 'fake',
    endpoint: 'http://localhost:4567'
});

kinesalite.listen(4567, function() {
    client.createStream({ ShardCount: 1, StreamName: 'one' }, function() {
        kinesalite.close(function() {
            kinesalite.listen(4567, function() {
                client.createStream({ ShardCount: 1, StreamName: 'one' }, function(err) {
                    if (err) throw err; // <---- this'll throw
                    kinesalite.close(function() {
                        console.log('made it!');
                    });
                });
            });
        });
    });
});
```

**Comment by mhart** [comment]
Bizarre error! But have reproduced no problem, thanks.

I'll look into it!

**Comment by mhart** [comment]
Oh right – I see, it's because you're trying to reuse the same kinesalite *instance*.

If you change this so that you create a new kinesalite instance each time it should work fine.

I'm not entirely sure that trying to reuse an instance once it's been closed should be a supported feature – it would mean hacking into the `listen` method and re-initializing the levelup database, etc – and `listen` isn't really a synonym for `open` so I don't think this is semantically correct.

I'll definitely make the error reporting a little nicer though.

**Comment by rclark** [comment]
Yeah the use-case is slightly odd, but common if you're using kinesalite in tests and you want to start fresh between individual tests.

I've been running tests like this with dynalite for some time, reusing the same instance. Is there a difference there?

**Comment by mhart** [comment]
@rclark there isn't a difference as of https://github.com/mhart/dynalite/issues/30

The problem is that `close` really should close everything, including the database – this is especially important when using an on-disk leveldb instance. To re-open the database, you need to create a new instance.

**Comment by mhart** [comment]
Didn't realise that a patch release would break this use case, so sorry about that! Problem is it's exploiting a bug that existed in the first place :smile_cat:

**Comment by rclark** [comment]
> Problem is it's exploiting a bug that existed in the first place

Totally fair. I guess it wouldn't be possible to detect and explicitly fail on reuse unless you hijack the `listen` method?

**Comment by mhart** [comment]
Yep, that's what I'm doing :+1:

**Comment by mhart** [comment]
Going with `Instance has been closed, please create a new instance to listen again` – sound ok?

**Comment by mhart** [comment]
Now that I'm looking at patching `listen`... I mean... I'm considering just recreating the store... :smiling_imp: 

I'll give it a quick test, and if it works maybe I'll just do it... If there are issues down the track with it, then can look at throwing an error instead.

**Comment by mhart** [comment]
Ok – so the only issue I've run into is with the 0 millisecond timeouts you're using on your create stream calls, etc.

Basically, I use `setTimeout` no matter what – so even with a 0 millisecond timeout, it can try to read/write after the database has closed.

I'm considering that a separate issue which I might try to patch

**Comment by mhart** [comment]
Ok, so I relented :smile_cat: – you should be able to keep using it as-is. Hopefully I haven't opened a can of worms! Don't really think so though...

**Comment by mhart** [comment]
Released as v1.4.3

**Comment by mhart** [comment]
Also fixed in dynalite, released as v0.11.3 – thanks for the report!
